### PR TITLE
Add QWebJS tests to the main test suite & fix tests

### PR DESCRIPTION
**Pull request by xmo-odoo**
* the qwebjs tests were not run as part of the regular suite
* some people *cough cough* had added tests which didn't passed in qwebpy but not qwebjs, some because missing features and others because tests covered underlying implementation details

### CODE REVIEW

File: addons/web/static/lib/qweb/tests.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
+odoo.define('qweb.tests', function () {
+'use strict';
+
+function trim(s) {
```
Comment (ged-odoo) [inline addons/web/static/lib/qweb/tests.js]: what is the difference between this and String.prototype.trim? (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/trim)

Comment (xmo-odoo) [reply addons/web/static/lib/qweb/tests.js]: I would expect none, this was just copied from the original file unmodified.

Comment (xmo-odoo) [reply addons/web/static/lib/qweb/tests.js]: PS: I sent you a mail yesterday, not sure you've seen it.

**Comment by Gorash** [comment]
https://github.com/odoo/odoo/blob/11.0/odoo/addons/base/tests/test_qweb.py#L491 ?

**Pull request closed without merging**